File: Array/bestTimetoBuyandSellStock.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
using namespace std;using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int minPrice = INT_MAX;
        int maxProfit = 0;
        
        for(int i=0; i<n; i++){
            minPrice = min(minPrice, prices[i]);
            maxProfit = max(maxProfit, prices[i]-minPrice);
        }
        
        return maxProfit;
    }
};

int main(){
    Solution sol;
    vector<int> prices = {7,1,5,3,6,4};
    cout << sol.maxProfit(prices) << endl;
    return 0;
}

File: SlidingWindow/bestTimeToBuyAndSellStock.py
def maxProfit(prices: list) -> int:
    min_price = float('inf')
    max_profit = 0

    for price in prices:
        if price < min_price:
            min_price = price
        else:
            max_profit = max(max_profit, price - min_price)

    return max_profit


if __name__ == "__main__":
    prices = list(map(int, input().split()))
    print("Max Profit:", maxProfit(prices))
